File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F14.P1.go
package main

import "fmt"

// BEGIN PROBE F14.P1
type Shape interface{ isShape() }

type Circle struct{ r float64 }

func (Circle) isShape() {}

type Rect struct{ w, h float64 }

func (Rect) isShape() {}

func newShape() Shape {
	var s Shape = Circle{2.0}
	return s
}

// END PROBE F14.P1

func main() {
	fmt.Println(newShape())
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F14.P1.rs
fn main() {
// BEGIN PROBE F14.P1
enum Shape { Circle { r: f64 }, Rect { w: f64, h: f64 } }
let s: Shape = Shape::Circle { r: 2.0 };
// END PROBE F14.P1
    if let Shape::Circle { r } = s {
        println!("{}", r);
    }
}
